### PR TITLE
[FLINK-12293][docs, datastream]Fix some comment typos in flink-streaming-java.

### PR DESCRIPTION
**Pull request by tianchen92**
## What is the purpose of the change

*This PR is related to [FLINK-12293](https://issues.apache.org/jira/browse/FLINK-12293). Fix some comment typos.*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BufferBlocker.java [hunk 1]
```diff
@@ -42,15 +42,15 @@
...
+	 * @return The readable sequence of buffers or events, or 'null', if nothing was added.
 	 */
 	BufferOrEventSequence rollOverWithoutReusingResources() throws IOException;
 
 	/**
 	 * Starts a new sequence of buffers and event reusing the same resources and
 	 * returns the current sequence of buffers for reading.
 	 *
-	 * @return The readable sequence of buffers and events, or 'null', if nothing was added.
+	 * @return The readable sequence of buffers or events, or 'null', if nothing was added.
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BufferBlocker.java]: This doesn't improve the doc. The sequence can contain both buffers **AND** events, not just one type of them.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BarrierBuffer.java [hunk 2]
```diff
@@ -64,7 +64,7 @@
 	/** The total number of channels that this buffer handles data from. */
 	private final int totalNumberOfInputChannels;
 
-	/** To utility to write blocked data to a file channel. */
+	/** The utility to write blocked data to a file channel. */
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/BarrierBuffer.java]: I am not a native speaker, but I think `A` utility would fit better here.

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java [hunk 3]
```diff
@@ -67,7 +67,7 @@
 
 	public static <X> ArrayKeySelector<X> getSelectorForArray(int[] positions, TypeInformation<X> typeInfo) {
 		if (positions == null || positions.length == 0 || positions.length > Tuple.MAX_ARITY) {
-			throw new IllegalArgumentException("Array keys must have between 1 and " + Tuple.MAX_ARITY + " fields.");
+			throw new IllegalArgumentException("The fields of array keys should between 1 and " + Tuple.MAX_ARITY + ".");
```
Comment (dawidwys) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/util/keys/KeySelectorUtil.java]: There is no verb in the sentence. Moreover how does it improve the original message?

**Comment by tianchen92** [comment]
@dawidwys Thanks for your comments, I have resolved them, please have a look.

**Comment by rmetzger** [comment]
Thank you for opening this PR to Flink. I would like to close this pull request, without merging it, as it does not substantially improve Flink.

Some of the fixed typos are indeed valid, while others are questionable. Still, these typos do not hurt any user experience (as the APIs are not user facing), nor do they make it harder to understand the APIs itself.
I'm sure we would converge to a state (after a few rounds of reviews) where this change is acceptable, but then we would have spend a lot of valuable committer/contributor time on a change not raising the bar for Flink.

**Comment by tianchen92** [comment]
@rmetzger Thanks for replying, I have closed this PR.

**Pull request closed without merging**